protocol/testutil/app/app.go

```diff
@@ -116,14 +116,19 @@ func DefaultTestApp(customFlags map[string]interface{}) *app.App {
 		nil,
 		true,
 		appOptions,
+		baseAppOptions...,
 	)
 	return dydxApp
 }
 
-// DefaultTestAppCreatorFn is a wrapper function around DefaultTestApp using the specified custom flags.
-func DefaultTestAppCreatorFn(customFlags map[string]interface{}) AppCreatorFn {
+// DefaultTestAppCreatorFn is a wrapper function around DefaultTestApp using the specified custom flags, and allowing
+// for optional base app options.
+func DefaultTestAppCreatorFn(
+	customFlags map[string]interface{},
+	baseAppOptions ...func(*baseapp.BaseApp),
```

Comment:
This looks like a good approach to allow more customization - could you get @lcwik to take a look since he has the most context here?

Reply:
This makes sense and LGTM.

It would be nice to instead pass in flags that set these on the `baseapp`. For example `FlagMinGasPrices`, this would allow us to test flags handling as well. This would be more substantive as it would require changing from `app.New` entry point to effectively something like a refactored [main()](https://github.com/dydxprotocol/v4-chain/blob/6cd6c0229ad5ec09a9be2faa4fa97778793bd1f8/protocol/cmd/dydxprotocold/main.go#L12).

Reply:
Ignoring this for now due to offline discussion.